src/group_theory/subgroup/basic.lean

```diff
@@ -1244,8 +1237,28 @@ subsingleton.elim _ _
 @[to_additive] lemma subgroup_of_bot_eq_top : H.subgroup_of ⊥ = ⊤ :=
 subsingleton.elim _ _
 
-@[simp, to_additive] lemma subgroup_of_self : H.subgroup_of H = ⊤ :=
-top_le_iff.mp (λ g hg, g.2)
+@[simp, to_additive] lemma subgroup_of_self : H.subgroup_of H = ⊤ := top_unique (λ g hg, g.2)
+
+@[simp, to_additive] lemma subgroup_of_inj {H₁ H₂ K : subgroup G} :
+  H₁.subgroup_of K = H₂.subgroup_of K ↔ H₁ ⊓ K = H₂ ⊓ K :=
+by simpa only [set_like.ext_iff, mem_inf, mem_subgroup_of, and.congr_left_iff] using subtype.forall
+
+-- TODO: left/right
+@[simp, to_additive] lemma subgroup_of_inf_left (H K : subgroup G) :
```

Comment:
Should it be `subgroup_of_inf_left`, `subgroup_of_inf_right`, or `inf_subgroup_of_right`?

Reply:
Probably `inf_subgroup_of_right`, I think.